### PR TITLE
feat: pass chain-id in prepareProposal

### LINKED ISSUE #1027: Pass ChainID via `RequestPrepareProposal`
the application will check signatures during prepare and process proposal, so we need to pass the chain-id in our requests since our version of the sdk doesn't have a mechanism to do that atm.

### PR DESCRIPTION
**Pull request by evan-forbes**
The cosmos-sdk doesn't have a good way to access the chain-id from the app atm, so we're passing via prepareProposal instead

closes #1027
part of fixing https://github.com/celestiaorg/celestia-app/issues/2020

### CODE REVIEW

File: proto/tendermint/types/types.proto [hunk 1]
```diff
@@ -211,17 +211,17 @@ message BlobTx {
...
-  repeated bytes data = 1;
-  repeated NMTProof share_proofs = 2;
-  bytes namespace_id = 3;
-  RowProof row_proof = 4;
-  uint32 namespace_version = 5;
+  repeated bytes    data              = 1;
+  repeated NMTProof share_proofs      = 2;
+  bytes             namespace_id      = 3;
+  RowProof          row_proof         = 4;
+  uint32            namespace_version = 5;
```
Comment (evan-forbes) [inline proto/tendermint/types/types.proto:218]: called proto lint + format, not sure why our protobuf linter doesn't also call those

**Comment by evan-forbes** [comment]
my protobuf setup adds an extra line `\n` in a few files relative to the CI, hence the protobuf generation failure.

either we can fix later, or someone with an identical setup to the CI can push to fix, but I don't think we should block on that given we need to ship asap

**Review by adlerjohn — COMMENTED** [review]
> either we can fix later, or someone with an identical setup to the CI can push to fix

1. Can you open an issue to look into this
2. Can you just manually edit the files to match? Otherwise all subsequent PRs will also have this check broken and actual bugs might slip through!

**Comment by evan-forbes** [comment]
> Can you just manually edit the files to match? Otherwise all subsequent PRs will also have this check broken and actual bugs might slip through!

@SweeXordious has a setup identical to CI, so this has fixed it. In the meantime I'll try and fix my local setup, or update the CI

File: proto/tendermint/abci/types.proto [hunk 2]
```diff
@@ -133,6 +133,8 @@ message RequestPrepareProposal {
   tendermint.types.Data block_data = 1;
   // If an application decides to populate block_data with extra information, they can not exceed this value.
   int64 block_data_size = 2;
+  // proposal_header is the header from the proposal
```
Comment (rootulp) [inline proto/tendermint/abci/types.proto:136]: [question] is this godoc copied from somewhere? If not, proposal to make the godoc reference the field name b/c I'm not sure what the relationship is between `proposal_header` and `chain_id` but they seem distinct. 

```suggestion
  // chain_id is a unique identifier for the blockchain network this proposal belongs to (i.e. mocha-1)
```

**Review by rootulp — APPROVED** [review]
one non blocking question about the godoc

File: proto/tendermint/abci/types.proto (hunk 2 above)
Comment (evan-forbes) [reply proto/tendermint/abci/types.proto:136]: I'll create an issue for this one, as changing will require recompiling the proto and the CI and my setup and different somehow despite using the same versions...

Comment (evan-forbes) [reply proto/tendermint/abci/types.proto:136]: #1029

**Review by SweeXordious — APPROVED** [review]

**Pull request merged**